### PR TITLE
Mes 2648/office page cpc

### PR DESCRIPTION
**Pull request by andrewsetterfield**
## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

## Screenshots (optional)

fail1
fail2
term
pass

### CODE REVIEW

File: src/modules/tests/test-summary/cat-cpc/test-summary.cat-cpc.reducer.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { createFeatureSelector } from '@ngrx/store';
+import * as fromCPCTestSummaryActions from './test-summary.cat-cpc.actions';
+import * as fromTestSummaryActions from '../common/test-summary.actions';
+import { TestSummary } from '@dvsa/mes-test-schema/categories/CPC';
+
+export const initialState : TestSummary = {
```
Comment (matthew2564) [inline src/modules/tests/test-summary/cat-cpc/test-summary.cat-cpc.reducer.ts:6]: initialState will need to have D255: false

File: src/pages/office/cat-cpc/components/combination/combination.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  @Input()
+  combination: Combination;
+
+  @Input()
+  combinationAdditionalText: string;
+
+  constructor() { }
+
+  // ngOnInit(): void {
```
Comment (matthew2564) [inline src/pages/office/cat-cpc/components/combination/combination.ts]: remove commented out code here and above

File: src/pages/office/cat-cpc/office.cat-cpc.page.html [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+                              [question5]="pageState.question5$ | async"
+                              [testOutcome]="outcome">
+            </cpc-debrief-card>
+
+          </div>
+        </ion-grid>
+      </ion-card-content>
+    </ion-card>
+
+    <div no-padding [hidden]="!(pageState.isTestOutcomeSet$ | async)">
```
Comment (matthew2564) [inline src/pages/office/cat-cpc/office.cat-cpc.page.html]: is this needed?

File: src/pages/office/cat-cpc/office.cat-cpc.page.ts [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+  getCombinationAdditionalText(code): string {
+    const question: Combination = questionCombinations.find((question) => {
+      return question.code === code;
+    });
+
+    return question.additionalText;
+  }
+
+  displayIfFail = (outcome: TestOutcome): boolean => {
+    console.log('outcome', outcome);
```
Comment (matthew2564) [inline src/pages/office/cat-cpc/office.cat-cpc.page.ts]: remove log

File: src/pages/office/cat-cpc/office.cat-cpc.page.ts [hunk 5]
```diff
@@ -0,0 +1,384 @@
...
+    this.store$.dispatch(new PersistTests());
+  }
+
+  getCombinationAdditionalText(code): string {
+    const question: Combination = questionCombinations.find((question) => {
+      return question.code === code;
+    });
+
+    return question.additionalText;
+  }
```
Comment (matthew2564) [inline src/pages/office/cat-cpc/office.cat-cpc.page.ts:287]: 'getCombinationAdditionalText(code: CombinationCodes): string {
    const question: Combination = questionCombinations.find((combination: Combination) => {
      return combination.code === code;
    });
    return question.additionalText;
  }
'

File: src/pages/office/cat-cpc/office.cat-cpc.page.ts [hunk 6]
```diff
@@ -0,0 +1,384 @@
...
+    };
+
+    const { testResult$, combination$ } = this.pageState;
+
+    this.subscription = merge(
+      testResult$.pipe(map(result => this.outcome = result)),
+      combination$.pipe(map(result => this.combinationAdditionalText = result)),
+    ).subscribe();
+
+    console.log('this.outcome', this.outcome);
```
Comment (matthew2564) [inline src/pages/office/cat-cpc/office.cat-cpc.page.ts]: remove log

File: src/pages/office/cat-cpc/office.cat-cpc.page.ts [hunk 7]
```diff
@@ -0,0 +1,384 @@
+import {
```
Comment (matthew2564) [inline src/pages/office/cat-cpc/office.cat-cpc.page.ts:1]: following discussions the other day can we group imports

File: src/pages/office/cat-cpc/office.cat-cpc.page.module.ts [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    IonicPageModule.forChild(OfficeCatCPCPage),
+    EffectsModule.forFeature([
+      OfficeAnalyticsEffects,
+      OfficeEffects,
+    ]),
+    ComponentsModule,
+    OfficeComponentsModule,
+  ],
+  providers: [
+    FaultSummaryProvider,
```
Comment (matthew2564) [inline src/pages/office/cat-cpc/office.cat-cpc.page.module.ts]: wouldnt have thought the FaultSummaryProvider is used in Office component for CPC?

File: src/pages/office/cat-cpc/office.cat-cpc.page.html [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+
+      <ion-card-content class="card-content-ios" no-bounce no-padding>
+        <ion-grid class="grid">
+          <activity-code id="activity-code-card" [formGroup]="form"
+                         [activityCodeModel]="pageState.activityCode$ | async"
+                         [activityCodeOptions]="activityCodeOptions"
+                         (activityCodeChange)="activityCodeChanged($event)" [disabled]="true">
+          </activity-code>
+
+          <!--          <div no-padding [hidden]="!(pageState.isTestOutcomeSet$ | async)">-->
```
Comment (matthew2564) [inline src/pages/office/cat-cpc/office.cat-cpc.page.html]: remove

File: src/pages/office/cat-cpc/components/combination/combination.ts [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+@Component({
+  selector: 'combination',
+  templateUrl: 'combination.html',
+})
+export class CombinationComponent {
+
+  // combinationAdditionalText: any;
+
+  @Input()
+  combination: Combination;
```
Comment (matthew2564) [inline src/pages/office/cat-cpc/components/combination/combination.ts]: not sure that is the correct interface, the interface for 'LGV1' for example is 'CombinationCodes'

File: src/pages/office/cat-cpc/components/assessment-report/assessment-report.ts [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+import { FormGroup, FormControl } from '@angular/forms';
+
+@Component({
+  selector: 'assessment-report',
+  templateUrl: 'assessment-report.html',
+})
+export class AssessmentReportComponent implements OnChanges {
+
+  @Input()
+  outcome: string;
```
Comment (matthew2564) [inline src/pages/office/cat-cpc/components/assessment-report/assessment-report.ts]: is the outcome used in the TS or HTML? if not please remove

Comment (andrewsetterfield) [reply src/pages/office/cat-cpc/components/assessment-report/assessment-report.ts]: i was using it to determine if the card was to be displayed prior to moving the logic to the office ts removed.

File: src/pages/office/cat-cpc/components/combination/combination.ts (hunk 2 above)
Comment (andrewsetterfield) [reply src/pages/office/cat-cpc/components/combination/combination.ts]: removed

File: src/modules/tests/test-summary/cat-cpc/test-summary.cat-cpc.reducer.ts (hunk 1 above)
Comment (andrewsetterfield) [reply src/modules/tests/test-summary/cat-cpc/test-summary.cat-cpc.reducer.ts:6]: added

File: src/pages/office/cat-cpc/office.cat-cpc.page.html (hunk 3 above)
Comment (andrewsetterfield) [reply src/pages/office/cat-cpc/office.cat-cpc.page.html]: based it off another office page, but no its not required.  removed.

File: src/pages/office/cat-cpc/office.cat-cpc.page.ts (hunk 4 above)
Comment (andrewsetterfield) [reply src/pages/office/cat-cpc/office.cat-cpc.page.ts]: removed

File: src/pages/office/cat-cpc/office.cat-cpc.page.ts (hunk 5 above)
Comment (andrewsetterfield) [reply src/pages/office/cat-cpc/office.cat-cpc.page.ts:287]: was having a hard time with that....so obvious.  had to make a few other changes to accommodate it. Done.

File: src/pages/office/cat-cpc/office.cat-cpc.page.ts (hunk 6 above)
Comment (andrewsetterfield) [reply src/pages/office/cat-cpc/office.cat-cpc.page.ts]: removed

File: src/pages/office/cat-cpc/office.cat-cpc.page.ts (hunk 7 above)
Comment (andrewsetterfield) [reply src/pages/office/cat-cpc/office.cat-cpc.page.ts:1]: just testing ya ;)

File: src/pages/office/cat-cpc/office.cat-cpc.page.module.ts (hunk 8 above)
Comment (andrewsetterfield) [reply src/pages/office/cat-cpc/office.cat-cpc.page.module.ts]: copy and paste monster....removed

File: src/pages/office/cat-cpc/office.cat-cpc.page.html (hunk 9 above)
Comment (andrewsetterfield) [reply src/pages/office/cat-cpc/office.cat-cpc.page.html]: removed

File: src/pages/office/cat-cpc/components/combination/combination.ts (hunk 10 above)
Comment (andrewsetterfield) [reply src/pages/office/cat-cpc/components/combination/combination.ts]: changed to match input

File: src/pages/office/cat-cpc/office.cat-cpc.page.ts [hunk 12]
```diff
@@ -0,0 +1,384 @@
...
+        select(getRekeyIndicator),
+        select(isRekey),
+      ),
+    };
+
+    const { testResult$, combination$ } = this.pageState;
+
+    this.subscription = merge(
+      testResult$.pipe(map(result => this.outcome = result as TestOutcome)),
+      combination$.pipe(map(result => this.combinationAdditionalText = result as CombinationCodes)),
```
Comment (matthew2564) [inline src/pages/office/cat-cpc/office.cat-cpc.page.ts]: the name 'combinationAdditionalText' is misleading, should be renamed to something like 'combinationCode'

Comment (andrewsetterfield) [reply src/pages/office/cat-cpc/office.cat-cpc.page.ts]: its the additional text from the combination, not the code...Ie 'Fire ex'....ah. its not at that point is it...fair point...i'll change it

Comment (matthew2564) [reply src/pages/office/cat-cpc/office.cat-cpc.page.ts]: its 'LGV1' for example, not the suffix, if it was the suffix this entire function is redundant getCombinationAdditionalText

Comment (andrewsetterfield) [reply src/pages/office/cat-cpc/office.cat-cpc.page.ts]: i think i edit'd whilst u were still writing...

File: src/pages/office/cat-cpc/__tests__/office.cat-cpc.page.spec.ts [hunk 13]
```diff
@@ -0,0 +1,415 @@
...
+    it('should not call navController when form is not valid', () => {
+      spyOn(component, 'isFormValid').and.returnValue(false);
+      spyOn(component.navController, 'push');
+      component.goToReasonForRekey();
+      expect(component.navController.push).not.toHaveBeenCalled();
+    });
+  });
+
+  describe('isFormValid', () => {
+    // it('should return true if form is valid', () => {
```
Comment (matthew2564) [inline src/pages/office/cat-cpc/__tests__/office.cat-cpc.page.spec.ts]: remove commented out code

File: src/pages/office/cat-cpc/__tests__/office.cat-cpc.page.spec.ts [hunk 14]
```diff
@@ -0,0 +1,415 @@
...
+    //   console.log('formValid', component.form.valid);
+    //   console.log('form', component.form);
+    //   expect(result).toEqual(true);
+    // });
+
+    it('should return true if form is valid', () => {
+      spyOn(component, 'createToast');
+      const form = component.form;
+      fixture.detectChanges();
+      // component.pageState.assessmentReport$ = of(true);
```
Comment (matthew2564) [inline src/pages/office/cat-cpc/__tests__/office.cat-cpc.page.spec.ts]: remove commented out code

File: src/pages/office/cat-cpc/__tests__/office.cat-cpc.page.spec.ts (hunk 13 above)
Comment (andrewsetterfield) [reply src/pages/office/cat-cpc/__tests__/office.cat-cpc.page.spec.ts]: removed

File: src/pages/office/cat-cpc/__tests__/office.cat-cpc.page.spec.ts (hunk 14 above)
Comment (andrewsetterfield) [reply src/pages/office/cat-cpc/__tests__/office.cat-cpc.page.spec.ts]: removed

File: src/pages/office/cat-cpc/components/combination/combination.html [hunk 15]
```diff
@@ -0,0 +1,17 @@
+<ion-row class="mes-validated-row mes-row-separator" id="combination-card">
+  <div class="validation-bar"></div>
+  <ion-col class="component-label" col-32 align-self-center>
+    <label>Combination</label>
+  </ion-col>
+  <ion-col align-self-center padding-left>
+    <ion-row class="spacing-row"></ion-row>
+    <ion-row>
+      <ion-col align-self-center>
+        <label id="combination-label">{{combination}} - {{combinationAdditionalText}}</label>
```
Comment (Cullima) [inline src/pages/office/cat-cpc/components/combination/combination.html]: think you need to handle combinations with no additional text or we will just get the '-' suffixed?

**Review by Cullima — CHANGES_REQUESTED** [review]

File: src/pages/office/cat-cpc/components/combination/combination.html (hunk 15 above)
Comment (andrewsetterfield) [reply src/pages/office/cat-cpc/components/combination/combination.html]: addressed

**Review by Cullima — APPROVED** [review]

**Review by matthew2564 — APPROVED** [review]

**Pull request merged**